cypress-io/cypress
# chore(deps): update to @faker-js/faker@8.4.1

- Closes [Update to @faker-js/faker 6.3.1 or later #30143](https://github.com/cypress-io/cypress/issues/30143)

### Additional details
-  Contributes to clearing Renovate's deprecation list on [Dependency Dashboard #3777](https://github.com/cypress-io/cypress/issues/3777).
- The function signature for `objectValue` is
```ts
objectValue<T extends Record<string, unknown>>(object: T): T[keyof T]
```
which appears to be the same return type as the deprecated `objectElement` method. A new type was created for the expected generic.

- The `arrayElement` method was moved to the [Helpers](https://fakerjs.dev/api/helpers.html#arrayelement) module and the `uuid` method was moved to the [String](https://fakerjs.dev/api/string.html#uuid) module.
- The `between`, `recent`, and `past` methods of the [Date](https://fakerjs.dev/api/date.html) module have all had their original function signatures deprecated and now require an `options` object to be passed.
- There may be [reproduction issues](https://fakerjs.dev/guide/usage.html#reproducible-results) related to the `date` module. Though, it doesn't appear that the current implementation will be effected, but referencing here for transparency.
- The `randomComponent` function seems to only be used [here](https://github.com/cypress-io/cypress/blob/develop/packages/frontend-shared/script/generate-stub-specs-ts.ts#L17-#L23) and writes to a file using the command `yarn workspace @packages/frontend-shared generate-stub-specs <fileName> <n> <baseName> [--app]` to generate fixtures. Refer to the [README](https://github.com/cypress-io/cypress/blob/develop/packages/frontend-shared/README.md#generating-fixtures) for the options.  This seems to work as intended if someone wants to verify.


### Steps to test
```
git clean -xfd
yarn
yarn workspace @packages/frontend-shared check-ts
yarn workspace @packages/frontend-shared cypress:run:ct
```

### How has the user experience changed?
No change.

### PR Tasks
- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
